buildshiprun/handler.go

```diff
@@ -470,82 +486,40 @@ func getEventFromEnv() (*sdk.Event, error) {
 	return &info, err
 }
 
-func functionExists(deploy deployment, gatewayURL string) (bool, error) {
-
-	r, _ := http.NewRequest(http.MethodGet, gatewayURL+"system/functions", nil)
-
-	addAuthErr := sdk.AddBasicAuth(r)
-	if addAuthErr != nil {
-		log.Printf("Basic auth error %s", addAuthErr)
-	}
-
-	res, err := http.DefaultClient.Do(r)
-
+func functionExists(ctx context.Context, client *faasSDK.Client, functionName string, gatewayURL string) (bool, error) {
+	// client := faasSDK.NewClient(&FaaSAuth{}, gatewayURL, nil, &timeout)
+	functions, err := client.ListFunctions(ctx, namespace)
 	if err != nil {
-		fmt.Println(err)
 		return false, err
 	}
 
-	defer res.Body.Close()
-
-	fmt.Println("functionExists status: " + res.Status)
-	result, _ := ioutil.ReadAll(res.Body)
-
-	functions := []function{}
-	json.Unmarshal(result, &functions)
-
 	for _, function1 := range functions {
-		if function1.Name == deploy.Service {
+		if function1.Name == functionName {
 			return true, nil
 		}
 	}
 
 	return false, err
 }
 
-func deployFunction(deploy deployment, gatewayURL string) (string, error) {
-	exists, err := functionExists(deploy, gatewayURL)
+func deployFunction(ctx context.Context, client *faasSDK.Client, deploySpec *faasSDK.DeployFunctionSpec, gatewayURL string) (string, error) {
+	var (
+		err error
+	)
+	exists, err := functionExists(ctx, client, deploySpec.FunctionName, gatewayURL)
 
-	bytesOut, _ := json.Marshal(deploy)
-
-	reader := bytes.NewBuffer(bytesOut)
-
-	fmt.Println("Deploying: " + deploy.Image + " as " + deploy.Service)
-	var res *http.Response
-	var httpReq *http.Request
-	var method string
+	fmt.Println("Deploying: " + deploySpec.Image + " as " + deploySpec.FunctionName)
```

Comment:
Should we use `log` here instead of `fmt`. I am aware that the code was here before, but just a note

Reply:
Replaced `fmt` by `log`